tests/infra/github.py

```diff
@@ -160,64 +162,53 @@ def get_latest_dev_tag(self):
         # Only consider tags that have releases as a release might be in progress
         return self._filter_released_tags(dev_tags)[0]
 
-    def get_release_branches_names(self, newest_first=False):
-        # Branches are ordered based on major version, with ordering based on newest_first
-        return sorted(
-            self.release_branches,
-            key=get_major_version_from_release_branch_name,
-            reverse=newest_first,
+    def get_tags_for_major_version(self, major_version=None):
+        version_re = f"{major_version}\." if major_version else ""
+        tag_re = f"^{TAG_RELEASE_PREFIX}{version_re}([.\d+]+)(-rc.*|)$"
+        tags = sorted(
+            [tag for tag in self.tags if re.match(tag_re, tag)],
```

Comment:
```suggestion
            tag for tag in self.tags if re.match(tag_re, tag),
```
Intermediary list isn't necessary.

Reply:
Python tells me that I need one:
```bash
32: Test command: /usr/bin/unbuffer "python3" "/home/jumaffre/git/CCF/tests/infra/github.py"
32: Test timeout computed to be: 10000000
32:   File "/home/jumaffre/git/CCF/tests/infra/github.py", line 169
32:     tag for tag in self.tags if re.match(tag_re, tag),
32:     ^
32: SyntaxError: Generator expression must be parenthesized
```

Reply:
```suggestion
            (tag for tag in self.tags if re.match(tag_re, tag)),
```

Should be fine?

Reply:
Sorry, that's my fault, I checked the generator expression worked in sorted() but as a sole argument, when more than one argument is used you do need `()`. But you do avoid the construction of the intermediary list still.